src/biokbase/narrative/jobs/appmanager.py

```diff
@@ -466,41 +415,281 @@ def run_app(
             return new_job
 
     @_app_error_wrapper
-    def run_app_bulk(self, app_info, cell_id=None, run_id=None, dry_run=False):
+    def run_app_bulk(
+        self,
+        app_info: list,
+        cell_id: str = None,
+        run_id: str = None,
+        dry_run: bool = False,
+    ) -> Union[dict, None]:
+        """
+        Attempts to run a batch of apps in bulk using the Execution Engine's run_app_batch endpoint.
+        If a cell_id is provided, this sends various job messages over the comm channel, and returns None.
+        If dry_run is True, this returns the structure that would be sent to EE2.run_job_batch
+
+        Parameters:
+        -----------
+        app_info: this is a list of app information dictionaries. It's broken down such that a single app
+            can have multiple sets of parameters, which could create multiple runs of that app.
+            Each dictionary is expected to have the following keys:
+            app_id: the id of the app to run
+            tag: the app tag to run, one of release, beta, or dev
+            version: (optional) the specified version to run, if not provided, this will be the most recent
+                for that particular tag
+            params: a list of at least dictionary. Each dict contains the set of parameters to run the
+                app once.
+        cell_id: if provided, this should be a unique id for the Narrative cell that's running the app.
+        run_id: if provided, this should be a unique id representing a Narrative cell's knowledge of
+            that job.
```

Comment:
What happens if these are not provided? Presumably it would not be possible to associate the job with a particular cell in the narrative.

Reply:
Yup, that's correct. It wouldn't be automatically associated with the cell.